### PR TITLE
Feature: Object States

### LINKED ISSUE #143: Add Hide function to GameObjects
A game developer may want to quickly/easily stop drawing an object without losing state information (for example, objects that you would like to see through but that maybe aren't actually clipped by the hardware yet).

Note: I'm not yet confident this makes sense as a GameObject function, since not all GameObjects are drawn anyways.  Maybe make a GameObject.disable(), and then it will not draw if it has a mesh, not light things if it contains a light, etc?  Feel free to suggest.

Current Use Case:  There are pipes on the walls in a current project.  Because of that, when viewing from outside, we'd like to clip the walls.  The flat surface of the wall is clipped by backface culling, but since the pipes are round, you see them through the walls.  It'd be preferred if we could just disable drawing on walls based on the camera direction.

### PR DESCRIPTION
**Pull request by dislogical**
This gives objects granular control over what updates each cycle, just like what DGame has.

Closes #143.

### CODE REVIEW

File: source/core/dgame.d [hunk 1]
```diff
@@ -23,7 +23,7 @@ enum EngineState
 /**
  * TODO
```
Comment (wozniakty) [inline source/core/dgame.d]: Someone (@ColdenCullen or he can suggest someone else who would know better) should do this TODO before we merge.

File: source/core/gameobject.d [hunk 2]
```diff
@@ -11,6 +11,34 @@ import std.conv, std.variant;
 
 enum AnonymousName = "__anonymous";
 
+shared struct ObjectUpdateFlags
```
Comment (wozniakty) [inline source/core/gameobject.d]: Ditto on documentation here

**Comment by ericchristenson** [comment]
Looks good to me now

**Comment by brooklynlittell** [comment]
:+1:

**Comment by wozniakty** [comment]
:+1:

**Pull request merged**